File: C Programming/problems on strings maipulation/problem108.c
/*

108.Write a program which accept string from user and accept one character. 
	 Return index of first occurrence of that character.
	 
	Input :   “Marvellous Multi OS”    M 
	Output :   0 
	Input :   “Marvellous Multi OS”    W 
	Output :   -1 
	Input :   “Marvellous Multi OS”    e 
	Output :   4 

	Author : Milind Nalawade
	
	Date : 16/07/2021
	
	Language : C programming
	
*/

#include<stdio.h>
#include<string.h>
int ChkChar(char* str,char ch)
{	
	int i=0;
	while(*str!='\0')
	{
		if (*str==ch)
		{
			break;
		}
		if(strlen(str)==i)
		{
			i=-1;
			break;
		}
		i++;
		str++;
	}
	return i;
}

//Entry point function
int main()
{
	char arr[20];
	char ch='\0';
	int ret=0;
	printf("\n Enter your string : ");
	scanf("%[^'\n']s",&arr);
	printf("\n Enter the character : ");
	scanf(" %c",&ch);
	ret=ChkChar(arr,ch);
	printf("\n First occurance of the character %c is : %d \n",ch,ret);
	return 0;
}	

File: Python Programming/problems on string manipulation/problem108.py
'''

108.Write a program which accept string from user and accept one character. 
	 Return index of first occurrence of that character..
	
	Input :   "Marvellous Multi OS"   M 
	Output :   0 
	Input :   "Marvellous Multi OS"    W 
	Output :   -1 
	Input :   "Marvellous Multi OS"    e 
	Output :   4 

	Author : Milind Nalawade
	
	Date : 16/07/2021
	
	Language : Python 3.9.0
	
'''
#====
#Logic
#====
def ChkChar(str,ch):
	j=1
	for i in str:
		if i==ch:
			break
		j+=1	
	if j==len(str)+1:
		j=-1	
	return j		
#=============
#Entry point function
#=============	
def main():
	str=input("\n Enter your character : ")
	ch=input("\n Enter you character : ")
	ret=int(ChkChar(str,ch))
	print("\n The first occurence of {}  in the given string is at place : {}\n".format(ch,ret))
	
#=====
#starter
#=====
if __name__=="__main__":
	main()
